### ISSUE TITLE
Meteor 1.4: cursor.count() wrong when using limit

### ISSUE THREAD

**Issue by veered** [meteor/meteor#7436]
On Meteor 1.4-beta.14, `SomeCollection.find({}, {limit: 1}).count()` will always return `1`, even if there is more than one document in `SomeCollection`. This is different than the behavior in Meteor 1.3. I've worked around this in the particular case I encountered it, but I expect it to cause issues elsewhere.

I believe this breaks the popular EasySearch package.

**Comment by abernix** [comment]
Thanks for reporting this!

Very simple reproduction:

#### 1.4-beta.14
```js
$ meteor shell
> Meteor.release
'METEOR@1.4-beta.14'
> const Test1 = new Meteor.Collection('test1');
> [1,2,3,4,5].forEach(v => Test1.insert({v}));
> Test1.find({}, {limit: 2}).count()
2
```

#### 1.3.4.4
```js
$ meteor shell
> Meteor.release
'METEOR@1.3.4.4'
> const Test1 = new Meteor.Collection('test1');
> [1,2,3,4,5].forEach(v => Test1.insert({v}));
> Test1.find({}, {limit: 2}).count()
5
```

@Fabs Do you have time to look at this?

**Comment by mitar** [comment]
I think this might be introduced with trying to fix the difference between server and client side? #1201 
M
aybe: bce4df9dfb0e135f2f01027b3b2a90030835e92c

(This change lacks test.)

**Comment by tmeasday** [comment]
We never did fix the inconsistency between client and server @mitar. All the commit above does is allow you to pass the `allowSkipLimit` through. I'll take a look at this now.

**Comment by tmeasday** [comment]
Ok, it seems like the mongo driver has switched its default behavior for `applySkipLimit`: https://jira.mongodb.org/browse/NODE-767

I guess I'll just default to `false` for now an add a test to ensure we don't break this in future.

**Comment by veered** [comment]
Awesome thanks!

**Comment by tmeasday** [comment]
See https://github.com/meteor/meteor/commit/258b14af0c68216506d3ffc6124331fc20a6f676

**Comment by benjamn** [comment]
This should be fixed if you do `meteor update --release 1.4-rc.1`.

**Comment by sferoze** [comment]
I just ran into this issue with Meteor 1.5.2

Using .count() on the server ignores the limit and skip values.

On the client, they are applied. 

While developing, it was a headache to debug, when you expect it to behave the same. 

@tmeasday why decide to default to `applySkipLimit` to `false` ?

The default is true on the client. So it would make sense for the default to be true for the server.

It is unexpected for them to be different and causes frustration in development. I had to search out this issue, '.count() different for client and server + meteor`
